packages/http/README.md

```diff
@@ -1,3 +1,376 @@
+
+## I want to make a http request a server
+
+In this use case we assume we have a server running at `http://localhost:4010`
+that is able to handle `GET /todos` request.
+
+We don't want to mock a reqeust, we simply want to make the request, hit the actual server and get the response back.
+
+```javascript
+const Prism = require('@stoplight/prism-http');
+
+// Create Prism instance and configure it to make http requests (mock: false)
+const config = { mock: false };
+const prism = Prism.createInstance(config);
+prism
+  .process({
+    method: 'get',
+    url: {
+      path: '/todos',
+      baseUrl: 'http://localhost:4010',
+    },
+  })
+  .then(prismResponse => {
+    console.log(prismResponse.output);
+  });
+```
+
+In response you'll get whatever the server responds with.
+
+## I want all my requests to talk to a server unless I override it per a reqeust
+
+In the following example we will first instantiate Prism to make requests to an actual server.
+
+Later we alter than behaviour by passing a config object to the `process` function.
+
+```javascript
+const Prism = require('@stoplight/prism-http');
+const path = require('path');
+
+// Note that by default we don't want to mock responses
+const prism = Prism.createInstance({ mock: false });
+const specPath = path.resolve(process.cwd(), 'basic.oas3.yaml');
+
+prism
+  .load({ path: specPath })
+  .then(() => {
+    // Make a "GET /todos" request
+    return prism.process(
+      {
+        method: 'get',
+        url: {
+          path: '/todos',
+          baseUrl: 'http://localhost:4010',
+        },
+      },
+      {
+        // We can override the default behaviour per request.
+        mock: {
+          dynamic: true,
+        },
+      }
+    );
+  })
+  .then(prismResponse => {
+    console.log(prismResponse.output);
+  });
+```
+
+# Advanced Topics
+
+## Creating Prism Instance
+
+In order to create and instance of Http Client (later referred to as `prism` for simplicity) use the `createInstance` function, like so:
+
+```javascript
+const Prism = require('@stoplight/prism-http');
+const config = { mock: false };
+const prism = Prism.createInstance(config /*, components */);
+```
+
+There are two (both optional) arguments you can supply `createInstance` with:
+
+- config (of `IHttpConfig` type)
+- components (of `TPrismHttpComponents` type)
+
+We will cover the `config` argument in next section and we'll leave `components` for some other time (overriding default `components` is the _ultimate advanced stuff_).
+
+### Config Property
+
+Prism's config object (`IHttpConfig`) allows you to manipulate Prism's behaviour in many ways. For instance:
+
+- turn validation on and off
+- turn mocking on and off
+- influence Prism's mocked response generation strategy
+
+The actual interface looks like this (but rather than explain each property I will give you some examples)
+
+```ts
+export interface IHttpConfig extends IPrismConfig {
+  mock: false | IHttpOperationConfig;
+
+  validate?: {
```

Comment:
This is stuff that non TypeScript people might struggle to understand.

Reply:
Any suggestions? The reason I copied this he is I didn't want to introduce yet another format for documenting our interfaces.

Reply:
Eh, not right now unfortunately. Maybe revisiting the API could help us to make it better, so I guess we'll keep it as it is right now.